### ISSUE TITLE
question about discriminative loss

### ISSUE THREAD

**Issue by VSJMilewski** [poojahira/image-captioning-bottom-up-top-down#24]
Hi @poojahira. Thanks for the codebase, it looks really nice and helps with my own implementations. 

I was wondering about the discriminative loss. in train.py you create the targets by iterating over the caption lengths and setting the target_d equal to target here in line [158](https://github.com/poojahira/image-captioning-bottom-up-top-down/blob/4fc5bf2980bc827602f42e0c23e82f1094d29603/train.py#L158). You keep overwriting the same values though. Shouldn't it be like this: 
`
for j, length in enumerate(decode_lengths):
            targets_d[j,:length-1] = targets[j,:length-1]
`

Furthermore, in other issues, you mention that you got the idea for this loss from other papers. Can you share which papers?

Thanks in advance!

**Comment by poojahira** [comment]
Hi, 
Please modify the loss function any way you want, it was just an experiment I tried. These are the papers I referred to:
https://arxiv.org/abs/1804.00887
https://arxiv.org/abs/1605.07912
There might be others as well.
